src-input/duk_bi_json.c

```diff
@@ -1423,7 +1423,11 @@ DUK_LOCAL void duk__enc_fastint_tval(duk_json_enc_ctx *js_ctx, duk_tval *tv) {
 	 * "long long" type exists.  Could also rely on C99 directly but that
 	 * won't work for older MSVC.
 	 */
+#if defined(__MINGW32__) || defined(__MINGW64__)
+	DUK_SPRINTF((char *) buf, "%I64d", (long long) v);
+#else
 	DUK_SPRINTF((char *) buf, "%lld", (long long) v);
+#endif
```

Comment:
Re: formatting more generally, internal code now usually casts to `unsigned long` or `long` and uses `%lu` and `%ld` because those work universally (also C89) for 32-bit values. There are no format specifier defines for exact 32-bit values etc (as there is in C99).

I didn't want to replicate all those defines (for C89) because there's relatively little need for them in practice besides debug prints (which are fine with a "wide cast" to long). Also, when using a sprintf() replacement, it's good that a minimal number of different specifiers are used in the format strings.

In concrete terms, there's exactly one non-debug call site where an "at least 64 bits" format specifier is needed - this fastint case - so a heavyweight fix seems wrong .. :)

There are some debug prints for fastint 64-bit values which may be truncated if sizeof(long) isn't at least 8 but that's not a functional issue. Also "long" is 64 bits on most Unix platforms (which are LP64) so debug prints there come out correctly.